pkg/asset/agent/gencrypto/auth_utils.go

```diff
@@ -11,3 +15,24 @@ func UserAuthHeaderWriter(token string) runtime.ClientAuthInfoWriter {
 		return r.SetHeaderParam("Authorization", token)
 	})
 }
+
+// ParseExpirationFromToken checks if the token is expired or not.
```

Comment:
Maybe in a followup PR, but we should try to avoid such code duplication and reuse the [assisted-service method](https://github.com/openshift/assisted-service/blob/8da29db1d92dea65758db3a3bd1919f8c630f2d0/internal/gencrypto/token.go#L98)

Reply:
I had this thought earlier, but in assisted the functions are in the internal package and will take more changes than just exporting a function. So I decided to replicate them in the installer.